### ISSUE TITLE
Mystery IP showing up in firewall

### ISSUE THREAD

**Issue by jameses986** [NebraLtd/helium-miner-software#99]
My setup....
pfsense firewall/router setup with a VLAN for separate traffic for miner.  One switch port on the SG-2100 is dedicated to the wired miner connection.  No other devices are on this VLAN, wireless, or wired connection.  POE injector is supplying power.
Miner is Skinny Blonde Eagle.
I cannot get it out of Relay mode and I believe it is because of the mysterious IP address that shows up about every 1-2 hours.
DHCP is on to hand out IP addresses, 44158 is wide open and verified with external port scan, and protocal flags have all be allowed to pass.  the miner is grabbing 172.17.0.2 within the DHCP server of 172.17.0.0/16.  I opened up the DHCP pool because I have noticed the other IPs showing.  The current mysterious IP is 172.17.0.3, and it is showing up as blocked by the native rules in my firewall.  I have added a rule to allow it to pass, and still waiting on the IP to be generated again...

See below for 2 images.  first one is showing all the pass rules for firewall on the VLAN.
second one is the mystery IP being blocked.  it is the only blocked traffic on the VLAN.  and I believe it is the reason I am still in Relay mode with miner,

Anyone else seeing the same in their firewall?  I am not sure what to do about it, and worry others are not aware of the situation.

![image](https://user-images.githubusercontent.com/87960050/126930618-1e400564-04e5-42f2-a32b-bbed70e54084.png)

![image](https://user-images.githubusercontent.com/87960050/126930477-95dd7608-e9c7-423b-9a54-1462d865849e.png)

**Comment by shawaj** [comment]
Have you got both WiFi and Ethernet connected?

That would be the only cause of two IP addresses.

**Comment by shawaj** [comment]
Then it's definitely not our unit. It only has two IP addresses if it is connected by both WiFi and Ethernet.

**Comment by shawaj** [comment]
@cwolfinger @jameses986 reopened in favour of the one on Helium miner GitHub as makes more sense to discuss here

**Comment by shawaj** [comment]
@vpetersson any thoughts on this? I'm guessing it's just the way the system is intentionally set up as the miner and diagnostics need to be exposed on the network. But wanted to get your opinion too.

**Comment by cwolfinger** [comment]
This seems to be a problem in the Docker runtime OR Linux.  Although following this thread it became more of a problem in the last 7 months --> https://github.com/docker/for-linux/issues/1126

**Comment by shawaj** [comment]
@cwolfinger IIRC you only have a bobcat miner right? but are experiencing the same issue?

@jameses986 could you email me the mac address, hotspot name, serial numbers of your unit to sales@nebra.com and let me know the ticket number? I will then see if I can dig into this further

**Comment by cwolfinger** [comment]
> @cwolfinger IIRC you only have a bobcat miner right? but are experiencing the same issue?
> 
> @jameses986 could you email me the mac address, hotspot name, serial numbers of your unit to [sales@nebra.com](mailto:sales@nebra.com) and let me know the ticket number? I will then see if I can dig into this further

Yes - that is why i think the issue is cross miners and not specific to Nebra.  The issue I believe is in the docker / linux side ( since others see the same problem with docker on different applications).

**Comment by luisflorencio** [comment]
+1
Nebra Outdoor / Firmware 2021.11.30.1-2

![Capturar](https://user-images.githubusercontent.com/59532805/147839702-7ca8dcba-4462-4211-bb29-02a63ea006da.PNG)

**Comment by tinkrlabs-io** [comment]
+1
Nebra Outdoor / Firmware 2022.01.04.0-2
![Capture](https://user-images.githubusercontent.com/95455290/148628972-aef24abc-19db-4458-b7f7-6e8e28e6c7fe.PNG)

**Comment by ericvitale** [comment]
+1
Nebra Outdoor / Firmware 2022.01.04.0-2

**Comment by shawaj** [comment]
This is a known issue but we aren't able to fix it currently as it's an issue with docker / balena engine. We have contacted then directly and also opened an issue here https://github.com/balena-os/balena-engine/issues/265

It's better to post on that issue than here

**Comment by ericvitale** [comment]
This is frustrating. My miner keeps switching between relayed = false to relayed = true and then back a couple times a day. I've done everything I can think of to fix this. The only thing I can think of is that this the problem.

**Comment by something-wrong** [comment]
Same problem with milesight miner

**Comment by shawaj** [comment]
> Same problem with milesight miner ![Screenshot ](https://user-images.githubusercontent.com/26740747/156896789-94bc4e44-f43f-4919-b102-8ef2943daddc.png)

This software has nothing to do with milesight, however as mentioned above this is an issue with docker

**Comment by shawaj** [comment]
> This is frustrating. My miner keeps switching between relayed = false to relayed = true and then back a couple times a day. I've done everything I can think of to fix this. The only thing I can think of is that this the problem.

I doubt it to be honest. This wouldn't effect anything to do with being relayed. Have you got the device connected to both WiFi and Ethernet perhaps?
